core/common/modules/finder/base-category.php

```diff
@@ -26,6 +27,26 @@ abstract class Base_Category extends Base_Object {
 	 */
 	abstract public function get_title();
 
+	/**
+	 * Get a unique category ID.
+	 *
+	 * TODO: Make abstract.
+	 *
+	 * @since 3.5.0
+	 * @access public
+	 *
+	 * @return string
+	 */
+	public function get_type() {
```

Comment:
That's not really deprecated, that's in order to inform the developers (via a PHP warning) that this function will be abstract in the future and they should implement it. (I added that with the help of @matipojo )

Reply:
if user created a category without override this class and another user create also category, the classes will override one of each other because they has the same `type`.
maybe in the implementation do: 'get_class($this )' to get the name of the class as the `type`

Reply:
Use always `get_id` that it's context less.
```suggestion
	public function get_id() {
```

Reply:
Done.